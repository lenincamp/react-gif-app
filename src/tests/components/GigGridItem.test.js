import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Tests in <GifGridItem />', () => {
  const title = 'Titulo';
  const url = 'https://localhost/algo.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('should show component right', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have paragraph with the title', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('should have the image equal to the url and alt of the props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('should have class animate__fadeIn', () => {
    const div = wrapper.find('div');
    expect(div.hasClass('animate__fadeIn')).toBe(true);
  });
});
