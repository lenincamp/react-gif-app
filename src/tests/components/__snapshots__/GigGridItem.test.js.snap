// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGridItem /> should show component right 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
